phantom-dsl/src/main/scala/com/websudos/phantom/CassandraTable.scala

```diff
@@ -189,25 +189,24 @@ abstract class CassandraTable[T <: CassandraTable[T, R], R] extends SelectTable[
     }
   }
 
-  Lock.synchronized {
+  val columns: Seq[AbstractColumn[_]] = Lock.synchronized {
 
-    val instanceMirror = cm.reflect(this)
     val selfType = instanceMirror.symbol.toType
 
-    // Collect all column definitions starting from base class
-    val columnMembers = MutableArrayBuffer.empty[Symbol]
-    selfType.baseClasses.reverse.foreach {
-      baseClass =>
-        val baseClassMembers = baseClass.typeSignature.members.sorted
-        val baseClassColumns = baseClassMembers.filter(_.typeSignature <:< ru.typeOf[AbstractColumn[_]])
-        baseClassColumns.foreach(symbol => if (!columnMembers.contains(symbol)) columnMembers += symbol)
-    }
-
-    columnMembers.foreach {
-      symbol =>
-        val column = instanceMirror.reflectModule(symbol.asModule).instance
-        _columns += column.asInstanceOf[AbstractColumn[_]]
-    }
+    val members: Seq[ru.Symbol] = (for {
+      baseClass <- selfType.baseClasses.reverse
+      symbol <- baseClass.typeSignature.members.sorted
+      if symbol.typeSignature <:< ru.typeOf[AbstractColumn[_]]
+    } yield symbol)(collection.breakOut)
+
+    for {
+      symbol <- members.distinct
+      table = if (symbol.isModule) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/flavian/phantom_2/file/2212686038/issues/source?bid=2847375&fileBranchId=3171525#l204)